src/OnboardingSPA/components/Drawer/DrawerPanel/DesignColors.js

```diff
@@ -1,15 +1,378 @@
+
+		if (colorStyle && selectedThemeColorPalette) {
+			for (let idx = 0; idx < selectedThemeColorPalette.length; idx++) {
+				switch (selectedThemeColorPalette[idx]?.slug) {
+					case 'primary':
+						if (isCustomStyle && primaryColorTemp)
+							selectedThemeColorPalette[idx].color = primaryColorTemp;
+						else
+							selectedThemeColorPalette[idx].color = colorPalettes[colorStyle][2];
+						break;
+					case 'secondary':
+						if (isCustomStyle && secondaryColorTemp)
+							selectedThemeColorPalette[idx].color = secondaryColorTemp;
+						else
+							selectedThemeColorPalette[idx].color = colorPalettes[colorStyle][1];
+						break;
+					case 'tertiary':
+						if (isCustomStyle && tertiaryColorTemp)
+							selectedThemeColorPalette[idx].color = tertiaryColorTemp;
+						else
+							selectedThemeColorPalette[idx].color = colorPalettes[colorStyle][0];
+						break;
+					case 'background':
+						if (isCustomStyle && backgroundColorTemp)
+							selectedThemeColorPalette[idx].color = backgroundColorTemp;
+						else
+							selectedThemeColorPalette[idx].color = '#ffffff';
+						break;
+				}
+			}
+
+			selectedGlobalStyle.settings.color.palette.theme = selectedThemeColorPalette;
+			setGlobalStyles(selectedGlobalStyle);
+			updatePreviewSettings(
+				useGlobalStylesOutput(selectedGlobalStyle, storedPreviewSettings)
+			);
+
+			return selectedGlobalStyle;
+		}
+	}
+
+	const getColorStylesAndPatterns = async () => {
+		const globalStyles = await getGlobalStyles();
+		let selectedGlobalStyle;
+		if (currentData?.data?.theme?.variation) {
+			selectedGlobalStyle = globalStyles.body.filter(
+				(globalStyle) =>
+					globalStyle.title === currentData.data.theme.variation
+			)[0];
+		} else {
+			selectedGlobalStyle = globalStyles.body[0];
+		}
+		setGlobalStyles(selectedGlobalStyle);
+
+		let selectedColors;
+		if (!currentData?.data?.palette[0]?.hasOwnProperty('supports')) {
+			currentData.data.palette[0] = {
+				"slug": "",
+       			"name": "",
+       			"color": [
+					{"slug": "primary", "name": "Primary", "color": ""},
+					{ "slug": "secondary", "name": "Secondary", "color": ""},
+					{ "slug": "tertiary", "name": "Tertiary", "color": ""},
+					{ "slug": "background", "name": "Background", "color": ""},
+				],
+				"supports": ["yith-wonder"]
+			};
+			selectedColors = currentData.data.palette[0];
+			setCurrentOnboardingData(currentData);
+		}
+		else {
+			selectedColors = currentData.data.palette[0];
+
+			if(selectedColors.slug === 'custom') {
+				setBackgroundColor(selectedColors?.color[3].color ?? null);
+				setPrimaryColor(selectedColors?.color[0].color ?? null);
+				setSecondaryColor(selectedColors?.color[1].color ?? null);
+				setTertiaryColor(selectedColors?.color[2].color ?? null);
+			}
+		} 
+		setSelectedColors(selectedColors);
+		setThemeColorPalette(currentData?.data?.palette[0]['slug'], selectedColors, selectedGlobalStyle);
```

Comment:
Again this is because there is a bug in the Flows due to which I can't change the design of the data structure and I have to stick with the existing one.